client/components/time-since/index.jsx

```diff
@@ -1,40 +1,36 @@
 import moment from 'moment';
-import { PureComponent } from 'react';
-import humanDate from 'calypso/lib/human-date';
-import { Interval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
-import smartSetState from 'calypso/lib/react-smart-set-state';
+import { useState, useEffect } from 'react';
+import getHumanDate from 'calypso/lib/human-date';
+import { useInterval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
 
-export default class TimeSince extends PureComponent {
-	smartSetState = smartSetState;
+function TimeSince( { className, date, dateFormat } ) {
+	const [ humanDate, setHumanDate ] = useState( '' );
+	const [ fullDate, setFullDate ] = useState( '' );
 
-	UNSAFE_componentWillMount() {
-		this.update();
-	}
+	function update( newDate ) {
+		const newHumanDate = getHumanDate( newDate ?? date, dateFormat );
+		const newFullDate = moment( newDate ?? date ).format( 'llll' );
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		this.update( nextProps.date );
-	}
+		if ( newHumanDate !== humanDate ) {
+			setHumanDate( newHumanDate );
+		}
 
-	update = ( date ) => {
-		const { dateFormat } = this.props;
-		date = date || this.props.date;
-
-		this.smartSetState( {
-			humanDate: humanDate( date, dateFormat ),
-			fullDate: moment( date ).format( 'llll' ),
-		} );
-	};
-
-	render() {
-		return (
-			<time
-				className={ this.props.className }
-				dateTime={ this.props.date }
-				title={ this.state.fullDate }
-			>
-				<Interval period={ EVERY_TEN_SECONDS } onTick={ this.update } />
-				{ this.state.humanDate }
-			</time>
-		);
+		if ( newFullDate !== fullDate ) {
+			setFullDate( newFullDate );
+		}
 	}
+
+	useInterval( update, EVERY_TEN_SECONDS );
+
+	useEffect( () => {
+		update( date );
+	}, [ update, date ] );
```

Comment:
The `update` dependency looks wrong because `update` is a new function on every render.